### DISCUSSION TITLE
Clarification about Jest?

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/turborepo#4295] [Help] [answered]
I've read the [testing docs page](https://turbo.build/repo/docs/handbook/testing), but it seems seriously lacking any info.

My setup is the same, I have `jest` installed in my root `package.json`. However every time I try to run `jest` I get:

```
command not found: jest
```

Where should I install Jest? Can I have some shared Jest config in its own package? Do I need to install Jest in each workspace within my monorepo? (installint it in the workspace makes it work but then it requires me to manage Jest for each workspace which isn't great)

If I move install `jest` in the specific workspace I'm getting a lot of:
```
Cannot find name 'expect'.
```

What should the type setup be?

**Comment by anthonyshew** [comment, marked as answer]
We now have a dedicated Jest page. Sorry for not having it sooner!
https://turbo.build/repo/docs/guides/tools/jest

**Reply by tgdn** [reply]
Assuming most of my packages will have tests, does it make more sense setting up jest in the root and using [projects [array<string | ProjectConfig>] ](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig) as described in [this SO answer](https://stackoverflow.com/questions/70999527/where-to-configure-jest-in-a-monorepo)?

**Reply by anthonyshew** [reply]
No, because then you can't cache tests per-package. Any time you make any change in the repository, you would miss cache for all tasks.
